bundles/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/discovery/OmnilinkDiscoveryService.java

```diff
@@ -406,22 +405,27 @@ private void discoverThermostats() {
 
                 Map<String, Object> properties = Map.of(THING_PROPERTIES_NAME, thingName);
 
-                switch (systemType.get()) {
-                    case LUMINA:
-                        thingUID = new ThingUID(THING_TYPE_LUMINA_AREA, bridgeUID, thingID);
-                        break;
-                    case OMNI:
-                        thingUID = new ThingUID(THING_TYPE_OMNI_AREA, bridgeUID, thingID);
-                        break;
-                    default:
-                        throw new IllegalStateException("Unknown System Type");
-                }
-
-                DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID).withProperties(properties)
-                        .withProperty(THING_PROPERTIES_NUMBER, thingID)
-                        .withRepresentationProperty(THING_PROPERTIES_NUMBER).withBridge(bridgeUID).withLabel(thingName)
-                        .build();
-                thingDiscovered(discoveryResult);
+                final String name = thingName;
+                systemType.ifPresentOrElse(t -> {
+                    ThingUID thingUID = null;
+                    switch (t) {
+                        case LUMINA:
+                            thingUID = new ThingUID(THING_TYPE_LUMINA_AREA, bridgeUID, thingID);
+                            break;
+                        case OMNI:
+                            thingUID = new ThingUID(THING_TYPE_OMNI_AREA, bridgeUID, thingID);
+                            break;
+                        default:
+                            throw new IllegalStateException("Unknown System Type");
+                    }
+                    DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID).withProperties(properties)
+                            .withProperty(THING_PROPERTIES_NUMBER, thingID)
+                            .withRepresentationProperty(THING_PROPERTIES_NUMBER).withBridge(bridgeUID).withLabel(name)
+                            .build();
+                    thingDiscovered(discoveryResult);
+                }, () -> {
+                    throw new IllegalStateException("Unknown System Type");
```

Comment:
Isn't it more reasonable to ignore or just log any unknown system type during discovery? Your binding will crash if it encounters an unknown system type with the current implementation.

Reply:
Sure, that works. I'll change it.

Reply:
Fixed @fwolter